### PR TITLE
partial fix for #9836

### PR DESCRIPTION
**Pull request by d-rothe**
This fixes that cleared error codes appear in DOOCS history where error
is not yet cleared.
It is only a partial fix since the history still contains more data than
was intended; intended was only first invalid data point

### CODE REVIEW

**Comment by d-rothe** [comment]
All unit tests are passing.
Also tested successfully with bcm_server, it shows intended behavior, error codes in history are correct now.

**Comment by d-rothe** [comment]
Please wait with review; just noticed I need to touch more files:
DoocsProcessArray.h etc have same problem

File: include/DoocsProcessScalar.h [hunk 1]
```diff
@@ -81,22 +81,11 @@ namespace ChimeraTK {
       //
       // We should also checked if data should be stored (flag storeInHistory). Invalid data should NOT be stored except first invalid data point.
```
Comment (killenb) [inline include/DoocsProcessScalar.h:82]: The whole comment can go away. The concept of not writing to history does not exist any more, and also storeInHistory is not there any more (commented code should be removed).

**Review by killenb — CHANGES_REQUESTED** [review]

File: include/DoocsProcessScalar.h [hunk 2]
```diff
@@ -47,11 +47,11 @@ namespace ChimeraTK {
       // DoocsUpdater
       auto data = _processScalar->accessData(0);
 
-      bool storeInHistory = true;
+      //bool storeInHistory = true;
       auto archiverStatus = ArchiveStatus::sts_ok;
       if(_processScalar->dataValidity() != ChimeraTK::DataValidity::ok) {
-        if(this->d_error()) // data are alredy invalid, do not store in history
-          storeInHistory = false;
+        //if(this->d_error())       // data are already invalid, do not store in history
```
[the remark this answers is not in the document]
Comment (d-rothe) [reply include/DoocsProcessScalar.h]: done

File: include/DoocsProcessScalar.h [hunk 3]
```diff
@@ -47,11 +47,11 @@ namespace ChimeraTK {
       // DoocsUpdater
       auto data = _processScalar->accessData(0);
 
-      bool storeInHistory = true;
+      //bool storeInHistory = true;
```
[the remark this answers is not in the document]
Comment (d-rothe) [reply include/DoocsProcessScalar.h]: done

**Review by killenb — APPROVED** [review]

**Pull request merged**